### PR TITLE
Add `x` directory and mutating webhook

### PR DESCRIPTION
**Pull request by alexmasi**

### CODE REVIEW

**Comment by chrisy** [comment]
Yikes, this has grown into quite the submission! Thanks for making this available!

**Comment by alexmasi** [comment]
I think I'm close, just hanging during topo creation. Logs from the webhook below:

```
alexmasi@alexmasi:~/kne/x/webhook$ k logs -l app=kne-assembly-webhook
2024/03/28 23:37:49 http: TLS handshake error from 10.244.0.1:1318: remote error: tls: bad certificate
2024/03/28 23:37:49 http: TLS handshake error from 10.244.0.1:28769: remote error: tls: bad certificate
2024/03/28 23:37:49 http: TLS handshake error from 10.244.0.1:6189: remote error: tls: bad certificate
2024/03/28 23:37:49 http: TLS handshake error from 10.244.0.1:52484: remote error: tls: bad certificate
```

@alshabib have you seen something like this before when developing? The genCerts.sh script is not documented and I am probably missing something there

**Comment by alshabib** [comment]
How was the webhook started? Was the certificate yaml also loaded?

**Comment by alexmasi** [comment]
> How was the webhook started? Was the certificate yaml also loaded?

The README describes the full process followed (which is an extension of the existing README internally)

File: x/webhook/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+the generic webhook along with an example mutator that simply adds an alpine
+linux container to created pods.
+
+To develop custom a custom mutation simply change the mutate function in the
+examples subdirectory.
+
+## Build
+
+```
+./containerize.sh
```
Comment (alshabib) [inline x/webhook/README.md:17]: This script isn't included here.

File: x/webhook/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+## Deployment
+
+For the webhook to be effective it must be deployed when the k8s cluster is up
+but before the KNE topology is created.
+
+Start by deploying the kubernetes cluster
+
+```
+$ kne deploy kne/deploy/kne/kind-bridge.yaml
```
Comment (alshabib) [inline x/webhook/README.md:28]: This should be `kne deploy ../../deploy/kne/kind-bridge.yaml`

Comment (alshabib) [reply x/webhook/README.md:28]: Assuming the command are called from the x directory.

**Comment by alshabib** [comment]
Ah yes sorry about that. When running `genCerts.sh` it updates  `tls.secret.yaml` but not the `mutating.config.yaml` where you need to copy the updated caBundle outputting by `genCerts.sh` to into `mutating.config.yaml`.

**Review by NehaManjunath — APPROVED** [review]

**Pull request merged**